src/mlpack/core/dists/diagonal_gaussian_distribution.hpp

```diff
@@ -18,15 +18,16 @@ namespace mlpack {
 namespace distribution {
 
 //! A single multivariate Gaussian distribution with diagonal covariance.
+template<typename MatType, typename Vectype>
```

Comment:
What is the reason to templatize the class at the same time? I guess I missed some things. I would probably opt for  `DiagonalGaussianDistributionType<...>` and provide an alias that uses `arma::mat`, because all of a sudden this looks a lot more intimidating to use.

Reply:
There are three reasons: first, this will remove partially the dependency on armadillo matrices, second, This will remove the need to inline all the functions from the class, and third, this will convert the class directly to a header implementation in order to make mlpack header only.